core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockletIndexFactory.java

```diff
@@ -110,9 +111,15 @@ public BlockletIndexFactory(CarbonTable carbonTable, IndexSchema indexSchema) {
    * @param carbonTable
    * @return
    */
-  public static Index createIndex(CarbonTable carbonTable) {
+  public static Index createIndex(CarbonTable carbonTable) throws IOException {
     boolean cacheLevelBlock = BlockletIndexUtil.isCacheLevelBlock(carbonTable);
-    if (cacheLevelBlock) {
+    boolean hasBloomIndex = false;
```

Comment:
better to have a check for cg datamap also in serialize data method..

If(blockcache && cg.isEmty) {dont write blocklet details} else {write blocklet details}

Reply:
Done. Added boolean check.